### PR TITLE
(#3108) - Document and add tests for opts.ajax.headers

### PR DESCRIPTION
**Pull request by daleharvey**

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -27,6 +27,8 @@ This method creates a database or opens an existing one. If you use a URL like `
 * `options.adapter`: One of `'idb'`, `'leveldb'`, `'websql'`, or `'http'`. If unspecified, PouchDB will infer this automatically, preferring IndexedDB to WebSQL in browsers that support both (i.e. Chrome, Opera and Android 4.4+).
 * `options.ajax`: An object of options to be sent to the ajax requester. In Node they are sent verbatim to [request][] with the exception of:
     * `options.ajax.cache`: Appends a random string to the end of all HTTP GET requests to avoid them being cached on IE. Set this to `true` to prevent this happening.
+* `options.ajax.headers`: The `ajax.headers` option allows you to customise headers that are sent to the remote HTTP Server.
+* `options.username` + `options.password`: You can specify HTTP auth parameters either by using a database with a name in the form `http://user:pass@host/name` or via the `username` + `password` options.
```
Comment (nolanlawson) [inline docs/api.md]: ^ this is actually wrong - it's `options.auth.username` and `options.auth.password` according to the code in http.js commented below

File: lib/adapters/http/http.js [hunk 2]
```diff
@@ -73,7 +73,7 @@ function getHost(name, opts) {
     uri.path = parts.join('/');
     opts = opts || {};
     opts = utils.clone(opts);
-    uri.headers = opts.headers || {};
+    uri.headers = opts.headers || (opts.ajax && opts.ajax.headers) || {};
 
     if (opts.auth || uri.auth) {
       var nAuth = opts.auth || uri.auth;
```
Comment (nolanlawson) [inline lib/adapters/http/http.js]: ^ This is where `username` and `password` come from. Is there any way to test this in your components test?

Comment (daleharvey) [inline lib/adapters/http/http.js]: Heh I did specifically add a test for it, just got the docs wrong

Comment (daleharvey) [inline lib/adapters/http/http.js]: Heres the test - https://github.com/pouchdb/pouchdb/pull/3418/files#diff-e71e1c5c6d0a286d583590ac933b89eeR35

Comment (nolanlawson) [inline lib/adapters/http/http.js]: cool cool, +1 after the docs fix

**Comment by daleharvey** [comment]
Merged in https://github.com/pouchdb/pouchdb/commit/e953ad45f48fdba2e4899a47442a66234b3fd2ad, thanks

**Pull request closed without merging**